src/main/java/de/tum/in/www1/artemis/repository/CompetencyRelationRepository.java

```diff
@@ -39,4 +39,52 @@ SELECT count(cr)
                     WHERE cr.headCompetency.course.id = :courseId OR cr.tailCompetency.course.id = :courseId
             """)
     long countByCourseId(@Param("courseId") long courseId);
+
+    @Query("""
+                    SELECT DISTINCT relation.headCompetency.id
+                    FROM CompetencyRelation relation
+                    LEFT JOIN relation.headCompetency
```

Comment:
nitpick but I would prefer some indentation for the left joins and the AND in the where clause